samples/java/hibernate/src/main/java/com/google/cloud/postgres/HibernateSampleTest.java

```diff
@@ -0,0 +1,260 @@
+    s.close();
+  }
+
+  public void testHqlList() {
+    Session s = hibernateConfiguration.openSession();
+
+    Query query = s.createQuery("from Singers");
+    List<Singers> list = query.list();
+    logger.log(Level.INFO, "Singers list size: {0}", list.size());
+
+    query = s.createQuery("from Singers order by fullName");
+    query.setFirstResult(2);
+    list = query.list();
+    logger.log(Level.INFO, "Singers list size with first result: {0}", list.size());
+
+    /* Current Limit is not supported. */
+    // query = s.createQuery("from Singers");
+    // query.setMaxResults(2);
+    // list = query.list();
+    // logger.log(Level.INFO, "Singers list size with first result: {0}", list.size());
+
+    query = s.createQuery("select  sum(sampleRate) from Tracks");
+    list = query.list();
+    logger.log(Level.INFO, "Sample rate sum: {0}", list);
+    s.close();
+  }
+
+  public void testOneToManyData() {
+    Session s = hibernateConfiguration.openSession();
+
+    Venues venues = s.get(Venues.class, UUID.fromString(venuesId.get(0)));
+    if (venues == null) {
+      logger.log(Level.SEVERE, "Previously Added Venues Not Found.");
+    }
+    if (venues.getConcerts().size() <= 1) {
+      logger.log(Level.SEVERE, "Previously Added Concerts Not Found.");
+    }
+
+    logger.log(Level.INFO, "Venues fetched: {0}", venues);
+    s.close();
+  }
+
+  public void testDeletingData() {
+    Session s = hibernateConfiguration.openSession();
+
+    Singers singers = Utils.createSingers();
+    s.getTransaction().begin();
+    s.saveOrUpdate(singers);
+    s.getTransaction().commit();
+
+    singers = s.get(Singers.class, singers.getId());
+    if (singers == null) {
+      logger.log(Level.SEVERE, "Added singers not found.");
+    }
+
+    s.getTransaction().begin();
+    s.delete(singers);
+    s.getTransaction().commit();
+
+    singers = s.get(Singers.class, singers.getId());
+    if (singers != null) {
+      logger.log(Level.SEVERE, "Deleted singers found.");
+    }
+    s.close();
+  }
+
+  public void testAddingData() {
+    Session s = hibernateConfiguration.openSession();
+    final Singers singers = Utils.createSingers();
+    final Albums albums = Utils.createAlbums(singers);
+    final Venues venues = Utils.createVenue();
+    final Concerts concerts1 = Utils.createConcerts(singers, venues);
+    final Concerts concerts2 = Utils.createConcerts(singers, venues);
+    final Concerts concerts3 = Utils.createConcerts(singers, venues);
+    s.getTransaction().begin();
+    s.saveOrUpdate(singers);
+    s.saveOrUpdate(albums);
+    s.saveOrUpdate(venues);
+    s.persist(concerts1);
+    s.persist(concerts2);
+    final Tracks tracks1 = Utils.createTracks(albums.getId());
+    s.saveOrUpdate(tracks1);
+    final Tracks tracks2 = Utils.createTracks(albums.getId());
+    s.saveOrUpdate(tracks2);
+    s.persist(concerts3);
+    s.getTransaction().commit();
+    s.clear();
```

Comment:
nit: I think you can remove this, as you are not using this session after this line anyways.

Reply:
Done